### DISCUSSION TITLE
free indexing for reading channel data

### DISCUSSION THREAD

**Discussion by szst11** [adamreeve/npTDMS#307] [Ideas]
Hallo,
as I understood the code, the indexing is only possible to as elipsis, silces or single values.
But for the use case of plotting a long time series, I migth take only a few samples for the plot. So from a file with many Gig of data I need only a limited number of resampled datapoints for the plot.
Another use case cold be a filtering, based on a complete loaded channel and load another channel with the filtering index.

**Comment by itrobinson** [comment]
This would be a really useful feature. It would provide a fast way to read a small selection of data from a large file.

I would like to be able to write something like this:

```python
import numpy
import nptdms
tdms_file = nptdms.TdmsFile.open('filename.tdms')
channel = tdms_file['group']['channel']
selection = numpy.linspace(0, len(channel), 1000, dtype='int', endpoint=False)
data = channel[selection]
```

In the code above `selection` is an array of 1000 elements evenly-spaced throughout the file.
